src/safeposix/syscalls/fs_calls.rs

```
@@ -785,43 +846,63 @@ impl Cage {
 
+                    // If the parent directory inode exists, make a reference of the oldpath inode
+                    // in the parent directory to make a link between the two directory paths.
                     (None, Some(pardirinode)) => {
+                        // Get the mutable instance of the parent inode object
                         let mut parentinodeobj =
                             FS_METADATA.inodetable.get_mut(&pardirinode).unwrap();
                         //insert a reference to the inode in the parent directory
                         if let Inode::Dir(ref mut parentdirinodeobj) = *parentinodeobj {
                             parentdirinodeobj
                                 .filename_to_inode_dict
                                 .insert(filename, inodenum);
+                            // Increment the link count of the parent inode as well because
+                            // when a link is created, a new directory entry is added to
+                            // the parent directory of the new link. 
                             parentdirinodeobj.linkcount += 1;
+                            //drop the mutable instance of the parent inode object
                             drop(parentinodeobj);
                             log_metadata(&FS_METADATA, pardirinode);
                             log_metadata(&FS_METADATA, inodenum);
                         } else {
+                            // If the parent inode is not of type "Directory", panic occurs.
                             panic!("Parent directory was not a directory!");
                         }
-                        0 //link has succeeded
+                        // If the linking is successful, 0 is returned.
+                        0
                     }
 
+                    // If the newpath exists, linking can't be perfomed and an error is returned.
                     (Some(_), ..) => syscall_error(Errno::EEXIST, "link", "newpath already exists"),
```

can reorder to fail early

---

Done